packages/trader/src/App/Components/Elements/PositionsDrawer/helpers/positions-helper.ts

```diff
@@ -27,7 +34,9 @@ export const getBarrierValue = (contract_info: TContractInfo) => {
     if (isDigitType(contract_info.contract_type)) {
         return digitTypeMap(contract_info)[contract_info.contract_type as keyof ReturnType<typeof digitTypeMap>];
     }
-    return addCommaToNumber(Number(contract_info.barrier));
+    return contract_info.barrier
+        ? formatMoney(contract_info.currency ?? 'USD', contract_info.barrier, true, 0, 0)
```

Comment:
@kate-deriv The number of the resulting decimal places here depends on the currency. But we should not rely here on currency at all. E.g. for different symbols, the number of decimal places will differ, and it has nothing to do with the currency. We should keep the same number of decimal places which we get in barrier string value. Please check  :)